lib/unifex/code_generator/base_types/enum.ex

```diff
@@ -92,6 +92,8 @@ defmodule Unifex.CodeGenerator.BaseTypes.Enum do
 
         #{if_statements}
 
+        unifex_free((void *) enum_as_string);
```

Comment:
Sorry I didn't notice it before. This way we won't free memory when `BaseType.generate_arg_parse` fails. If `BaseType.generate_arg_parse` fails it will goto exit label and it will omit this unifex_free. I am not sure how to fix it. Should we write parsing atom directly here instead of using `BaseType.generate_arg_parse`? 
One more thing, before freeing memory we shoould check if it's not equal to NULL so
```c
char * enum_as_string = NULL;
....
if (enum_as_string != NULL) {
  unifex_free...
}
```

@mat-hek

Reply:
That's right. I see two ways to solve that:
- execute `BaseTypes.Atom.generate_arg_parse` manually
- declare `enum_as_string` in `generate_initialization` (it will have to have a unique name then) and clean it up in `generate_destruction`